app/gui/view/component-browser/component-list-panel/grid/src/entry/style.rs

```diff
@@ -97,75 +176,72 @@ pub struct Style {
-        color: &frp::Stream<color::Lcha>,
-        color_intensities: &frp::Stream<ColorIntensities>,
+        main_color: &frp::Stream<color::Lcha>,
+        colors: &frp::Stream<Colors>,
         is_dimmed: &frp::Stream<bool>,
     ) -> Self {
-        fn mix((c1, c2): &(color::Lcha, color::Lcha), coefficient: &f32) -> color::Lcha {
-            color::mix(*c1, *c2, *coefficient)
-        }
-        let app_bg = style_watch.get_color(ensogl_hardcoded_theme::application::background);
-        let color_intensities = color_intensities.clone_ref();
+        let bg = style_watch.get_color(panel_theme::background_color);
+        let colors = colors.clone_ref();
+        let color_anim = color::Animation::new(network);
 
-        let intensity = Animation::new(network);
         frp::extend! { network
             init <- source_();
 
-            text_intensity <- color_intensities.map(|c| c.text);
-            bg_intensity <- color_intensities.map(|c| c.background);
-            hover_hg_intensity <- color_intensities.map(|c| c.hover_highlight);
-            dimmed_intensity <- color_intensities.map(|c| c.dimmed);
-            icon_strong_intensity <- color_intensities.map(|c| c.icon_strong);
-            icon_weak_intensity <- color_intensities.map(|c| c.icon_weak);
-
-            one <- init.constant(1.0);
+            bg_intensity <- colors.map(|c| c.background_intensity);
+            bg <- all_with(&bg, &init, |col, ()| color::Lcha::from(col));
+            bg_and_main <- all(&bg, &color_anim.value);
             let is_dimmed = is_dimmed.clone_ref();
-            intensity.target <+ is_dimmed.switch(&one, &dimmed_intensity);
-            app_bg <- all_with(&app_bg, &init, |col, ()| color::Lcha::from(col));
-            app_bg_and_input <- all(&app_bg, color);
-            main <- app_bg_and_input.all_with(&intensity.value, mix);
-            app_bg_and_main <- all(&app_bg, &main);
-            background <- app_bg_and_main.all_with(&bg_intensity, mix).sampler();
-            hover_highlight <- app_bg_and_main.all_with(&hover_hg_intensity, mix).sampler();
-            text <- app_bg_and_main.all_with(&text_intensity, mix).sampler();
-            icon_weak <- app_bg_and_main.all_with(&icon_weak_intensity, mix).sampler();
-            icon_strong <- app_bg_and_main.all_with(&icon_strong_intensity, mix).sampler();
+            dimmed <- all_with3(&init, main_color, &colors,
+                |_, main, colors| colors.dimmed.resolve(main)
+            );
+            color_anim.target <+ switch(&is_dimmed, main_color, &dimmed);
+
+            // Component list panel does not support semi-transparent background. Instead, we mix
+            // the color of the component browser's background and the main color of the
+            // component group.
+            background <- all_with(&bg_and_main, &bg_intensity,
+                |(bg, main), intensity| color::mix(*bg, *main, *intensity)
+            ).sampler();
```

Comment:
Again, I don't understand this. `Component list panel does not support semi-transparent background` - but here we compute the color of grid view entry, not component list panel?

Reply:
Yep, I messed up the wording. In fact, we don't support semi-transparent backgrounds either for entries or for panels. But here it means the entries' background. Reworded it.